File: Battleship/Battleship.Logic/Factories/PlayerFactory.cs

using Battleship.Logic.Factories.Interfaces;
using Battleship.Logic.InternalModels;
using Battleship.Logic.InternalModels.Interfaces;

namespace Battleship.Logic.Factories
{
    public class PlayerFactory : IPlayerFactory
    {
        public IPlayer GetPlayer(int playerId)
        {
            if (playerId == 1)
            {
                return new Player1();
            } 
            else
            {
                return new Player2();
            }
        }
    }
}

File: Battleship/Battleship/ClientApp/src/app/factories/player-factory.ts
import { Injectable } from "@angular/core";
import { GameModel } from '../models/game-model';
import { IPlayer } from '../models/interfaces/player-interface';
import { Player1 } from '../models/player1-model';
import { Player2 } from '../models/player2-model';

@Injectable({
  providedIn: 'root'
})
export class PlayerFactory {
  getOtherPlayerFromGame(playerId: number, game: GameModel): IPlayer {
    if (playerId === 1) {
      return new Player2(2);
    } else if (playerId === 2) {
      return new Player1(1);
    }
  }

  getThisPlayerFromGame(playerId: number, game: GameModel): IPlayer {
    if (playerId === 1) {
      return new Player1(1);
    } else if (playerId == 2) {
      return new Player2(2);
    }
  }
}
